services/models/pod.go

```diff
@@ -93,3 +93,20 @@ func lookupImage(containerName string, containers []v1.Container) string {
 	}
 	return ""
 }
+
+func (pod *Pod) HasIstioSideCar() bool {
+	return len(pod.IstioContainers) > 0
+}
+
+func (pods Pods) HasIstioSideCar() bool {
+	results := []bool{}
```

Comment:
That intermediate slice looks unnecessary, no? You could just have a single for-loop over pods, and return false at the first `pod.HasIstioSideCar()` being false.

Reply:
YI knwo...I talked with @xeviknal  about and the posibility of have some pod with istiosidecar and others with not, before if the first pod has istiosidecar we were assuming that all pods have istioSidecar but There could be some without

Reply:
yes but still you can do that:

```
func (pods Pods) HasIstioSideCar() bool {
    results := []bool{}
    for _, pod := range pods {
        if !pod.HasIstioSideCar() {
           return false;
        }
    }
    return true
}
```

Reply:
Indeed. You can have both two loops combined into one.
Another thing is that it changes the original behavior. I would leave it as before since this flag is used to determine if that workload/service has istio information in it. I imagine the workload/service list, and I would like see which item has istio in it and thus all the istio information.
In case there is one pod that doesn't have the sidecar deployed, we will show a warning/mark next to it in the workload details in order to warn users about that.

Reply:
IMO, yes. And it is what we have in service list.
If not, we shouldn't reuse this method with service list unless we want to change the behavior there too.